htdocs/PI/write/PIWriteRequest.php

```diff
@@ -743,6 +699,108 @@ private function updateEntity(Site $siteService) {
         $this->httpResponseCode = 204;
     }
 
+    /**
+    * Takes the URL and turns it into a useful array
+    *
+    * Also checks some things and tidy's it up
+    *@param string $url A section of the url the user accessed the wirte API
+    *@throws \exception
+    *@return array $requestArray an array of useful things extracted from the url
+    */
+    private function processURLtoArray($url) {
+      #If the request isn't set then no url parameters have been used
+      if (is_null($url)) {
+          $this->httpResponseCode=400;
+          throw new \Exception($this->genericExceptionMessages["URLFormat"]);
+      }
+
+      #Split the request into seperate parts, with the slash as a seperator
+      #Note that apache will collapse multiple /'s into a single /)
+      $requestArray = explode("/",$url);
+
+      #We should probably ignore trailing slashes, which will generate an empty array element
+      #Using strlen so that '0' is not removed
+      $requestArray = array_filter($requestArray, 'strlen');
+
+      #The request url should have either 4 or 5 elements
+      if(!in_array(count($requestArray),array(4,5))){
+          $this->httpResponseCode=400;
+          throw new \Exception(
+              "Request url has the wrong number of elements. " . $this->genericExceptionMessages["URLFormat"]
+          );
+      }
+
+      return $requestArray;
+    }
+
+    /**
+    *Processes the API Version
```

Comment:
Should this block have spaces after the `*`?

Reply:
Yes. This and 11 similar lines are fixed in the latest version